ConsoleApp_Task1/Program.cs

```diff
@@ -82,44 +34,68 @@ public static double ReturnRadiusOrSide()
                             {
                                 InputString = InputString.Replace(",", ".");
                             }
-                            else
-                            {
-
-                            }
-                            double RadiusOrSideTest = Convert.ToDouble(InputString);
-                            var CircleOrArea = new Figure(RadiusOrSideTest) { };
-                            Console.WriteLine("\nInput Value is " + CircleOrArea.input);
-                            return CircleOrArea.input;
+                            _input = Convert.ToDouble(InputString);
+                            Console.WriteLine("Input Value is " + _input);
+                            return _input;
                         }
                     }
-
                     catch (SystemException sex)
                     {
                         // this class's error string
                         string LastError = sex.Message;
-                        Console.WriteLine("\nIncorrect Input: value is not parsed correctly");
+                        Console.WriteLine("Incorrect Input: value is not parsed correctly");
                         e++;
                     }
                 }
                 else
                 {
-                    Console.WriteLine("\nIncorrect Input: Empty value is not allowed");
+                    Console.WriteLine("Incorrect Input: Empty value is not allowed");
                     e++;
                 }
             }
-
             if (e == 3)
             {
                 double minNumber = 0.5;
                 double maxNumber = 5.0;
-                double RadiusOrSide = new Random().NextDouble() * (maxNumber - minNumber) + minNumber;
-                Figure OtherCircleOrArea = new Figure(RadiusOrSide) { };
-                Console.WriteLine("\nRandom value is " + Math.Round((OtherCircleOrArea.input),2));
-                return OtherCircleOrArea.input;
-
+                _input = new Random().NextDouble() * (maxNumber - minNumber) + minNumber;
+                Console.WriteLine("\nRandom value is " + Math.Round((_input), 2));
```

Comment:
this is the candidate for the bug: you set some random value to the input, but show to the user the rounded one.
Later you will calculate the area based on the input and the result can be different for the rounded and not rounded values.
Isn't it better to set the rounded value to input right away?